drivers/lcd/lcd_framebuffer.c

```diff
@@ -173,7 +173,30 @@ static int lcdfb_updateearea(FAR struct fb_vtable_s *vtable,
   fb_coord_t endy;
   int ret;
 
-  DEBUGASSERT(area != NULL);
+  if (area == NULL)
+    {
+      run  = priv->fbmem;
+
+      for (row = 0; row < priv->yres; row++)
+        {
+          ret = pinfo->putrun(pinfo->dev, row, 0, run, priv->xres);
+          if (ret < 0)
+            {
+              return ret;
+            }
+
+          run += priv->stride;
+        }
+
+      /* Just redraw the display 
+       * TODO add new function pointer to xtruct lcd_planeinfo_s
+       * Now calling pointer putrun with npix = 0 activates redrawing display  
+       */
+      
+      ret = pinfo->putrun(pinfo->dev, 0, 0, priv->fbmem, 0);
```

Comment:
if last argument has value 0 driver redraws display. Maybe it is better to add a pointer to the function that redraws display?

Reply:
Do you notice putarea callback in lcd_planeinfo_s? you don't need additional putrun with special parameters to notify lcd driver that the update is end.

Reply:
@adamkaliszan could you review whether https://github.com/apache/incubator-nuttx/pull/6564 could resolve your problem.

Reply:
#6564 solve the problem partially. It solve the problem of NULL value, but two things are missing:
1) caling funcion **ret = pinfo->putrun(pinfo->dev, 0, 0, priv->fbmem, 0);**
2) Eink driver changes. I can add to my pull request code from #6564. My version has less operation of assignments, but some part of code is duplicated. 

I would like to discuss callling function  **ret = pinfo->putrun(pinfo->dev, 0, 0, priv->fbmem, 0);** If there is 0 pixels to update, the e-ink driver redraws the screen. Maybe it is better to add (to object **pinfo**) a pointer to the function that triggers redrawing screen. 
In order to reduce the number of changes I decided to use pointer **putrun** to trigger redrawing the screen.

I can merge https://github.com/apache/incubator-nuttx/pull/6564 to my pull request. 

Unfortunately SSD1680 driver has no implemented function updatearea. I can implement it in a future. This display has very nice support for it, when works in portrait mode.

Reply:
> #6564 solve the problem partially. It solve the problem of NULL value, but two things are missing:
> 
> 1. caling funcion **ret = pinfo->putrun(pinfo->dev, 0, 0, priv->fbmem, 0);**
> 2. Eink driver changes. I can add to my pull request code from [drivers/lcd_framebuffer: Optimize updateearea for the full screan case #6564](https://github.com/apache/incubator-nuttx/pull/6564). My version has less operation of assignments, but some part of code is duplicated.
> 
> I would like to discuss callling function **ret = pinfo->putrun(pinfo->dev, 0, 0, priv->fbmem, 0);** If there is 0 pixels to update, the e-ink driver redraws the screen. Maybe it is better to add (to object **pinfo**) a pointer to the function that triggers redrawing screen. In order to reduce the number of changes I decided to use pointer **putrun** to trigger redrawing the screen.
> 

if lcd driver doesn't implement updatearea, the additional callback or special meaning of putrun in lcd_dev_s is required to get the notification of the end update. I think the new callback is better because:

1. it is more clean to express the intent
2. it is more compatible with the driver which doesn’t need the end of update.

> I can merge #6564 to my pull request.
> 
> Unfortunately SSD1680 driver has no implemented function updatearea. I can implement it in a future. This display has very nice support for it, when works in portrait mode.

Reply:
I agree. I will add new callback and modify ssd1680 e-ink driver.